csaf/src/model/store.rs

```diff
@@ -4,7 +4,15 @@ use std::path::{Path, PathBuf};
 
 /// create a distribution base directory
 pub fn distribution_base(base: impl AsRef<Path>, distribution: &Distribution) -> PathBuf {
-    base.as_ref().join(
-        utf8_percent_encode(distribution.directory_url.as_str(), NON_ALPHANUMERIC).to_string(),
-    )
+    if let Some(_directory_url) = &distribution.directory_url {
+        base.as_ref().join(
+            utf8_percent_encode(
+                distribution.directory_url.clone().unwrap().as_str(),
```

Comment:
Please don't ignore the outcome of the let binding and then perform an .unwrap().